plugins/stylescombo/plugin.js

```diff
@@ -109,14 +109,41 @@
 				},
 
 				onRender: function() {
+					function getCutOffIndex(elements, tagNames) {
+						// given an element path and an array of tag names to stop on,
+						// returns the earliest index of any of the tag names, or -1
+
+						var cutOffIndexes = [];
+						var elementNames = elements.map(function(element) {
```

Comment:
I wonder if you could accomplish this same thing with a object instead, where the key is the element name and the value is the earliest index for that element. Then below, you could get that index in constant time.

I think you'd have to iterate over the elements in reverse with the index.